services/vcpkg/vcpkg-version.tester.js

```diff
@@ -1,11 +1,23 @@
-import { isSemver } from '../test-validators.js'
+import Joi from 'joi'
 import { createServiceTester } from '../tester.js'
 
 export const t = await createServiceTester()
 
-t.create('gets the port version of entt')
+t.create('gets the port version of entt using `version` field')
   .get('/entt.json')
-  .expectBadge({ label: 'vcpkg', message: isSemver })
+  .expectBadge({ label: 'vcpkg', color: 'blue', message: Joi.string() })
```

Comment:
**note**: Related to the [other comment](https://github.com/badges/shields/pull/8945/files#r1117987743), in order to avoid having tests failing if the version pattern changes (e.g. date format is changed to semantic versioning), we only test if the version has been properly extracted by looking at the color of the badge.

Reply:
How about we keep one "live" test and test this level of detail by either:

- splitting out the logic for choosing a version field into its own function and unit testing it (have a look for `*.spec.js` files in `services/` for examples) or
- using mocked tests for each detail case (have a look for `*.tester.js` that include `.intercept` files in `services/` for examples)

If we pick a project that uses the `version-semver` field and validate with `isSemver` for the integration test, that should be stable enough.

Reply:
I went for the second approach and mocked the other test cases for the different version fields in https://github.com/badges/shields/pull/8945/commits/23a169acbe210b80c06a5f62945b7d1707a60f45. Let me know if this makes sense!

Reply:
Since I ran into some issues mocking these tests, I went for the approach of extracting the version parsing helper.

The remaining live test is checking a port that currently exposes its version through the `version-semver` as you suggested.